scripts/common.py

```diff
@@ -712,3 +715,99 @@ def __enter__(self):
 
     def __exit__(self, etype, value, traceback):
         os.chdir(self.saved_path)
+
+
+@contextlib.contextmanager
+def managed_process(command_args, shell=False, timeout_secs=60, **kwargs):
+    """Context manager for starting and stopping a process gracefully.
+
+    Args:
+        command_args: list(int | str). A sequence of program arguments, where
+            the program to execute is the first item.
+        shell: bool. Whether the command should be run inside of its own shell.
+        timeout_secs: int. The time allotted for the managed process and its
+            descendants to terminate themselves. After the timeout, any
+            remaining processes will be killed abruptly.
+        **kwargs: dict(str: *). Same kwargs as `subprocess.Popen`.
+
+    Yields:
+        psutil.Process. The process managed by the context manager.
+    """
+    if PSUTIL_DIR not in sys.path:
+        sys.path.insert(1, PSUTIL_DIR)
+    import psutil
+
+    # We're only using these generators to reduce code duplication.
+    stripped_args = (python_utils.UNICODE(a).strip() for a in command_args)
```

Comment:
Not sure about the need for the UNICODE cast. Shouldn't all args be ascii anyway? If acceptance of unicode chars is needed, did you test that they're handled correctly? (I don't see a test for it.)

Reply:
I just did it this way because post-Python 3 migration this will become `str(a).strip()`. The unicode-ness isn't really important here, it's for casting ints into strings.